src/executables/util_sa.c

```diff
@@ -1716,10 +1716,29 @@ diagdb (UTIL_FUNCTION_ARG * arg)
   if (diag == DIAGDUMP_ALL || diag == DIAGDUMP_HEAP)
     {
       bool dump_records;
-      /* this dumps the contents of all heaps */
+      const char *class_name;
       dump_records = utility_get_option_bool_value (arg_map, DIAG_DUMP_RECORDS_S);
-      fprintf (outfp, "\n*** DUMP OF ALL HEAPS ***\n");
-      (void) file_tracker_dump_all_heap (thread_p, outfp, dump_records);
+      class_name = utility_get_option_string_value (arg_map, DIAG_CLASS_NAME_S, 0);
+      if (class_name == NULL)
+	{
+	  /* this dumps the contents of all heaps */
+	  fprintf (outfp, "\n*** DUMP OF ALL HEAPS ***\n");
+	  (void) file_tracker_dump_all_heap (thread_p, outfp, dump_records);
+	}
+      else
+	{
+	  /* this dumps the contents of a specific heap for given class name */
+	  if (!sm_check_system_class_by_name (class_name))
+	    {
+	      /* check if the format of class name is valid */
+	      if (utility_check_class_name (class_name) != NO_ERROR)
+		{
+		  goto error_exit;
+		}
+	    }
+	  fprintf (outfp, "\n*** DUMP OF SPECIFIC HEAP(S) ***\n");
```

Comment:
To be kinder, it would be better to include the class name entered with the --class-name option in the heading.

*** DUMP HEAP OF user_name.class_name ***